misc/emacs/ql-mode-base.el

```diff
@@ -0,0 +1,74 @@
+;;; ql-mode-base.el --- A major mode for editing QL files
+
+;;; Commentary:
+;;
+;; A basic major mode for editing QL files, a more advanced major
+;; mode is available for internal use at Semmle.
+;;
+;; Provides syntax highlightning, comment support, and a mode-specific
+;; keymap.
+
+;;; Code:
+
+(defconst ql--at-type-regex "\\_<@\\w+\\>")
+(defconst ql--predicate-regex "\\(\\_<\\w+\\(\\+\\|\\*\\)?\\_>\\)\\s-*(")
+(defconst ql--primitive-type-regex (regexp-opt '("int" "string" "float" "boolean" "date") 'symbols))
+(defconst ql--annotation-regex (regexp-opt '("abstract" "cached" "external" "final" "transient" "library" "private" "deprecated" "override" "query") 'words))
+(defconst ql--parameterized-annotation-regex "\\<\\(pragma\\|language\\|bindingset\\) *\\[ *[a-z, ]*\\]")
```

Comment:
I think that `a-z` should also allow uppercase for `monotonicAggregate` to work.

It's slightly unfortunate that `bindingset[this]` and `bindingset[result]` are not highlighted correctly, although I'm not sure how important you feel that is.

Reply:
Ok. I have simplified it. It is now consistent  with Vim.

Example display when loaded in `emacs -q`: 
![image](https://user-images.githubusercontent.com/42067045/52854624-49001480-311f-11e9-9fa2-277ddfcf2510.png)

Reply:
Great, this looks good to me locally.